go/libraries/doltcore/sqle/dsess/session.go

```diff
@@ -1234,8 +1234,10 @@ func setPersistedValue(conf config.WritableConfig, key string, value interface{}
 	}
 }
 
-// SystemVariablesInConfig returns system variables from the persisted config
-func SystemVariablesInConfig(conf config.ReadableConfig) ([]sql.SystemVariable, error) {
+// SystemVariablesInConfig returns system variables from the persisted config.
+// It ignores persisted keys that do not exist in |sql.SystemVariables|. The
+// ignored keys are sent to |cb|.
+func SystemVariablesInConfig(conf config.ReadableConfig, cb func(key string)) ([]sql.SystemVariable, error) {
```

Comment:
i think you need to pick one, a callback that takes `func(sql.SystemVariable, error)` or a signature that returns `valid, inValid []sql.SystemVariable`. A combo of the two is more difficult for a generic user even when the doc comment is good.